plugin/src/main/java/org/opensearch/ml/plugin/MachineLearningPlugin.java

```diff
@@ -247,6 +249,12 @@ public class MachineLearningPlugin extends Plugin implements ActionPlugin, Searc
 
     private ConversationalMemoryHandler cmHandler;
 
+    private volatile boolean ragSearchPipelineEnabled;
+
+    public MachineLearningPlugin(Settings settings) {
+        this.ragSearchPipelineEnabled = MLCommonsSettings.ML_COMMONS_RAG_PIPELINE_FEATURE_ENABLED.get(settings);
+    }
```

Comment:
This is not required. You can always read the settings directly. one example: https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/KNNSettings.java#L449-L455, similar example should be there in this repo too. Let me try to find it.

Reply:
This example doesn't apply as it is tied to an index.

Reply:
Found a better example from ML commons plugin: https://github.com/opensearch-project/ml-commons/blob/f88e18c5873c9d7e198f1137b7269364bcd5c1bd/plugin/src/main/java/org/opensearch/ml/settings/MLFeatureEnabledSetting.java#L15-L34

Reply:
Yes, this is already done.